5] 2001/main.py

```diff
@@ -0,0 +1,43 @@
+import re
+import operator
+import random
+
+
+def simulate(code):
+    """Calculate dice roll from dice with formula: xDy+z where:
+    y – type of dice to use (e.g. D6, D10),
+    x – number of dice rolled; for a single roll this parameter is omitted,
+    z – modifier - number to add (or subtract) to the result of the roll (optional).
+
+    Args:
+        code (string): the code entered for the simulation
+
+    Raises:
+        ValueError: given code is invalid
+
+    Returns:
+        int: result of simulation
+    """
+    types = [3, 4, 6, 8, 10, 12, 20, 100]
+    operations = {'+': operator.add, '-': operator.sub,
+                  '*': operator.mul, '/': operator.truediv}
+    try:
+        x = re.findall('\d+(?=D)', code)
```

Comment:
tohle je opravdu super slozite a velmi problematicke na cteni. Takhle slozite se to nedela to prasovani pres regexy - mel by pouzit grupy

Reply:
no tohle bylo proto, abych když vyparsuju třeba / tak abych to mohl použít hned ve výpočtu pro return a nemusel jsem dělat if operations je / tak to udělej tak atd. Když tam může být jenom + a nebo - tak to je o dost jednodušší.